extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx

```diff
@@ -58,6 +60,9 @@ export function RemoteQueries(): JSX.Element {
     return <div>Waiting for results to load.</div>;
   }
 
+  const [repoListExpanded, setRepoListExpanded] = useState(false);
```

Comment:
I'm still not sure what `useState(...)` means 😵 Did you find any useful explanations about it?

(Don't worry if not, I'm willing to accept that it does the right thing either way 😂)

Reply:
Quick reference since we were discussing this offline: https://reactjs.org/docs/hooks-state.html.

The idea here is that we declare a new variable `repoListExpanded` that becomes part of the active "state" of the component, together with a setter function `setRepoListExpanded` that we call every time we want to update the value (instead of mutating the variable directly). The argument to `useState` (`false` here) is the initial value of the variable.

I don't work with React that often, so would love to hear how Charis (and others!) think about React state too.

Reply:
Yep...that's right. This helps the react engine know exactly what parts of the UI need redrawing and refreshing. If the state of a component changes, then it needs to be refreshed (and all child components). Other components do not need to be redrawn.

Reply:
Thanks everyone!

@shati-patel please resolve the conversation if you're happy or grab me to discuss further if not! :)